aya/src/programs/fentry.rs

```diff
@@ -0,0 +1,78 @@
+//! fentry programs.
+use thiserror::Error;
+
+use crate::{
+    generated::{bpf_attach_type::BPF_TRACE_FENTRY, bpf_prog_type::BPF_PROG_TYPE_TRACING},
+    obj::btf::{Btf, BtfError, BtfKind},
+    programs::{load_program, utils::attach_btf_id, LinkRef, ProgramData, ProgramError},
+};
+
+#[derive(Debug)]
+#[doc(alias = "BPF_TRACE_FENTRY")]
+#[doc(alias = "BPF_PROG_TYPE_TRACING")]
+pub struct FEntry {
+    pub(crate) data: ProgramData,
+}
+
+#[derive(Debug, Error)]
+pub enum FEntryLoadError {
+    #[error(transparent)]
+    Btf(#[from] BtfError),
+
+    #[error(transparent)]
+    Program(#[from] ProgramError),
+}
+
+/// Marks a function as a fentry eBPF program that can be attached to almost
+/// any function inside the kernel. The difference between fexit and kretprobe
+/// is that fexit has practically zero overhead to call before kernel function.
+/// fentry programs can be also attached to other eBPF programs.
+///
+/// # Minimumm kernel version
+///
+/// The minimum kernel version required to use this feature is 5.5.
+///
+/// # Examples
+///
+/// ```no_run
+/// # #[derive(thiserror::Error, Debug)]
+/// # enum Error {
+/// #     #[error(transparent)]
+/// #     BtfError(#[from] aya::BtfError),
+/// #     #[error(transparent)]
+/// #     FEntry(#[from] aya::programs::FEntryLoadError),
+/// #     #[error(transparent)]
+/// #     Program(#[from] aya::programs::ProgramError),
+/// #     #[error(transparent)]
+/// #     Bpf(#[from] aya::BpfError),
+/// # }
+/// # let mut bpf = Bpf::load_file("ebpf_programs.o")?;
+/// use aya::{Bpf, programs::FEntry, BtfError, Btf};
+/// use std::convert::TryInto;
+///
+/// let btf = Btf::from_sys_fs()?;
+/// let program: &mut FEntry = bpf.program_mut("filename_lookup").unwrap().try_into()?;
+/// program.load("filename_lookup", &btf)?;
+/// program.attach()?;
+/// # Ok::<(), Error>(())
+/// ```
+impl FEntry {
+    /// Loads the program inside the kernel.
+    ///
+    /// See also [`Program::load`](crate::program::Program::load).
+    ///
+    /// # Arguments
```

Comment:
We follow the rust std doc style, which doesn't have an Arguments section. Instead please document what the arguments do in discursive form, eg:

```
Loads the program so it's executed when the kernel function `fn_name` is entered. The `btf` argument must contain the BTF info for the running kernel.
```

Reply:
Done